libs/brig-types/src/Brig/Types/User.hs

```diff
@@ -235,6 +235,45 @@ instance FromJSON SelfProfile where
 instance ToJSON SelfProfile where
     toJSON (SelfProfile u) = toJSON u
 
+----------------------------------------------------------------------------
+-- Rich info
+
+data RichInfo = RichInfo
+    { richInfoFields :: ![RichField]
+    }
+    deriving (Eq, Show)
+
+instance ToJSON RichInfo where
+    toJSON u = object
+        [ "fields" .= richInfoFields u
+        , "version" .= (0 :: Int)
+        ]
+
+instance FromJSON RichInfo where
+    parseJSON = withObject "RichInfo" $ \o -> do
+        o .:? "version" >>= \case
+            Nothing -> RichInfo <$> o .: "fields"
+            Just (0 :: Int) -> RichInfo <$> o .: "fields"
+            Just v -> fail ("unknown version: " <> show v)
+
+data RichField = RichField
+    { richFieldType  :: !Text
+    , richFieldValue :: !Text
+    }
+    deriving (Eq, Show)
+
+instance ToJSON RichField where
+    toJSON u = object
+        [ "type" .= richFieldType u
```

Comment:
`"name"` instead of `"type"`?  or `"key"`?  i like both better than `"type"`.

Reply:
I also like "name" better, but having "type" in SCIM gives us more hope for compatibility because `{"type": ..., "value": ...}` is how all other SCIM stuff is formatted, and it's quite possible that some provisioning agent would support `"type"` but not `"name"`. Given this, having `"type"` in SCIM but `"name"` in Brig would be unnecessarily confusing.

Reply:
Ok, agreed, but that would be good to document.  Just a short sentence like "'type' is better than the laternative 'key' or 'name' because SCIM uses it in analogous places."